### ISSUE TITLE
Comparison

### ISSUE THREAD

**Issue by joelgeorgem** [jp-praveen/Rendezvous#23]
Use the data in Table 3 of the second paper that we told we will read. Run our greedy algorithm on it. Compare the results.

**Comment by jp-praveen** [comment]
Sir, Actually the asteroids orbit the sun, so should I give the same frame as given in the paper and input Gravitational parameter of the sun? I tried to convert orbital elements of the Asteroids as given in Zheng et.al. 2015, to the ECI frame but for that, we need more information about the orbit.

**Comment by joelgeorgem** [comment]
I did not notice that the data is for asteroids. Yes, use for &mu; the value that corresponds to Sun. Everything else remains the same. If you want the initial and final positions and velocities directly, then use the corresponding table in the paper that I have mailed you. And of course, since the time scales are different, use higher time steps.

**Comment by jp-praveen** [comment]
Sir, I made use of the positions and velocities from the paper you sent and the orbital elements from Zheng et.al for orbit propagation. I am encountering some errors and currently, I am rectifying them. I will inform you soon when the code is fully working.

**Comment by jp-praveen** [comment]
Sir, 
I used the radius and velocity values given in the paper that you sent. But in order to propagate the asteroids position and velocity over time, I needed time period and specific angular momentum of the asteroids which I calculated using the eccentricity and mean anomaly data from Zheng et.al paper. I took the chaser's initial position as given in Ahn & Bang. I ran the greedy algorithm for 2 cases,
1. Transfer time goes from 50 days to 1500 days with 50 days as time step.
2. Transfer time goes from 50 days to 1500 days with 10 days as time step.

Results for case 1:
1. sequence_best =  3     4     2     1
2. dv = 24.4836 km/s
3. total time = 1.0282e+9 seconds
 
Results for case 2:
1. sequence_best =  3     4     2     1
2. dv = 19.6797 km/s
3. total time = 1.1794e+09 seconds

The paper got the best sequence as 1 2 3 4 with a dv of 6.36 km/s.
Looking at Table 5, from Ahn and Bang, it is clear that they have been coasting for atleast 2000 days before making a transfer and from the debris data we know that coasting can reduce the total dv. So I tried to use the coasting time they have used but the chaser's initial orbit's parameters are not given in the paper and I was not able to propagate the orbit.

**Comment by joelgeorgem** [comment]
(r,v) is equivalent to orbital parameters. There will be subroutines available that will do this; or you can write one on your own.

**Comment by jp-praveen** [comment]
Sir, I used the (r,v) values and included the conversion formulas to necessary orbital elements in our code itself. I ran the code for the 1st case in table 5 of Ahn and Bang paper. It is the transfer from chaser to the 1st asteroid with 2033.48 days as coasting time and 169.08 days as transfer time. We got dv=1.658 km/s and the paper got a dv of 1.53 km/sec.


- [ ] time

**Comment by joelgeorgem** [comment]
Coasting seems to be important for asteroid rendezvous problem. Our initial studies showed that coasting does not help much in debris removal problem. However, the study was done before we got a "correct" code for Lambert transfer. Can we repeat the study now and see whether the result still holds?

**Comment by jp-praveen** [comment]
Sir, Actually our initial study showed that coasting did not help in minimizing the transfer time. Later we found that dv minimum does depend on coasting time and for different coasting times we get a different and sometimes a better dv min.  I also ran the corrected code now whose results are as below,

coasting time = 139968 secs  dv_min=0.0592 km/s
![1](https://user-images.githubusercontent.com/55296086/71587024-2eba0400-2b42-11ea-8e78-46b91b17e93c.jpg)

coasting time = 189968 secs  dv_min=0.3639 km/s
![2](https://user-images.githubusercontent.com/55296086/71587039-3d082000-2b42-11ea-8060-9121daf28ba7.jpg)

coasting time = 239968 secs   dv_min=0.8665 km/s
![3](https://user-images.githubusercontent.com/55296086/71587060-4a250f00-2b42-11ea-9ffc-e3b700348eab.jpg)

coasting time = 289968 secs   dv_min=0.4610 km/s
![4](https://user-images.githubusercontent.com/55296086/71587072-50b38680-2b42-11ea-92b6-d5230ba62b26.jpg)
9968 secs

coasting time = 439968 secs   dv_min=0.9044 km/s
![5](https://user-images.githubusercontent.com/55296086/71587098-67f27400-2b42-11ea-879b-3ac8843b7336.jpg)

**Comment by joelgeorgem** [comment]
For the above case, what is the dv_min for zero coasting time?

**Comment by jp-praveen** [comment]
Sir, For zero coasting time and for the same range of transfer time, dv_min  is 1.9594 km/sec.

**Comment by joelgeorgem** [comment]
Then we should consider coasting time also while computing &Delta;V. We should pre-compute _extremal field maps_ and use them as lookups in our algorithm.

**Comment by jp-praveen** [comment]
Yes Sir. I will read about the extremal field maps. Do you have any good source from where I can learn about them as I am not exposed to them previously?

**Comment by joelgeorgem** [comment]
I guess EFM is just a fancy name for a 3D plot of &Delta;V as a function of coasting time and transfer time. The term appears in publications by Junkins at TAMU.

**Comment by jp-praveen** [comment]
Yes Sir. When I tried to search for some good sources to learn I ended up getting only journals. I will make a 3D plot (dv_min, transfer_time, coasting_time) and will send it shortly.

**Comment by joelgeorgem** [comment]
Not dv_min, but dv. A 3D plot of (transfer_time, coasting_time, dv). We will compute dv_min from this.

**Comment by jp-praveen** [comment]
Yes sir, I am making some modifications to the code. I will finish them and plot the graph.

**Comment by jp-praveen** [comment]
Sir, I encountered some problems (like memory and dimension errors) while making the grid points and it took me some time to make clean data (without nan and zero values) and rectify the errors. I did a 3-D mesh for the transfer between debris 2 to debris 14. The coasting time varies from 5000 secs to 150000 secs with an increment of 50 secs. For each coasting time, the transfer time varies from 500 secs to 10000 sec with an increment of 50 secs.

![1](https://user-images.githubusercontent.com/55296086/71645429-68376e80-2cfe-11ea-9dc2-36d24e56761a.jpg)

![4](https://user-images.githubusercontent.com/55296086/71645646-37a50400-2d01-11ea-8dc5-0ecf94b1d2bf.jpg)

**Comment by joelgeorgem** [comment]
Can you clip the max value of dv to 1 and replot the figures? For example, if dv is the 3D matrix containing the data, then do ``` dv(dv>1)=1; ``` and plot dv.

**Comment by jp-praveen** [comment]
Sir, The figures are attached below. 

![1](https://user-images.githubusercontent.com/55296086/71676364-19451400-2da6-11ea-881c-4ed14c8303e6.jpg)
![2](https://user-images.githubusercontent.com/55296086/71676365-19ddaa80-2da6-11ea-8aea-ea73032681d7.jpg)
![3](https://user-images.githubusercontent.com/55296086/71676367-1a764100-2da6-11ea-9f69-38a343f4a0a7.jpg)
![4](https://user-images.githubusercontent.com/55296086/71676368-1a764100-2da6-11ea-9376-98fb6a51b52d.jpg)

**Comment by jp-praveen** [comment]
Sir, The paper with the debris data is using Lambert Rendezvous with drift and for the transfer from debris 2 to debris 14 they used: Coasting time = 139968s, transfer time=2592s, dv=0.0598 km/s. 

I ran our Lambert algorithm with a stepsize of 1sec for coasting time and got the minimum transfer as: Coasting time = 139892s, transfer time = 8400s, dv=0.0571 km/s. This dv is lower than than the dv in the paper.

**Comment by joelgeorgem** [comment]
Why is it that the transfer time in your simulations is starting at 4000?

**Comment by jp-praveen** [comment]
Sir, I did the simulation with the number of revolutions set to 1 (since it took a long time for the code to run if I took all 3 revolution case). In this transfer case, I randomly checked some points throughout the range of coasting time and transfer time and the dv was minimum only for the m=1 case. For m=1, usually the lower transfer time returns no dv, that's why there are no data points before 4000s.

**Comment by jp-praveen** [comment]
Sir,
     I wrote the Particle Swarm Algorithm (particle_swarm.m) for our problem. Initially, I took 10000 points and 1000 iterations. But the global minimum is obtained in less than 80 iterations. In this way, it is quite faster to obtain the minimum dv.
